src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -48,6 +48,13 @@ public bool CanNavigateToSpan(Workspace workspace, DocumentId documentId, TextSp
 
             var document = workspace.CurrentSolution.GetDocument(documentId);
```

Comment:
how one can guarantee TextSpan given is belong to CurrentSolution snapshot? either API needs to change to accept Document and span that belong to the Document snapshot, and regardless this crash, this implementation should always make sure given span is valid.